Added quick start section and workflow description. (#9)
Arm-Examples/AVH-Hello: README.md

## Quick Start

1. Install [Keil Studio for VS Code](https://marketplace.visualstudio.com/items?itemName=Arm.keil-studio-pack) from the VS Code marketplace.
2. Clone this repository (for example using [Git in VS Code](https://code.visualstudio.com/docs/sourcecontrol/intro-to-git)) or download the ZIP file. Then open the base folder in VS Code.
3. Open the [CMSIS View](https://mdk-packs.github.io/vscode-cmsis-solution-docs/userinterface.html#2-main-area-of-the-cmsis-view) in VS Code and use the ... menu to choose an example via *Select Active Solution from workspace*.
4. The related tools and software packs are downloaded and installed. Review progress with *View - Output - CMSIS Solution*.
5. In the CMSIS view, use the [Action buttons](https://github.com/ARM-software/vscode-cmsis-csolution?tab=readme-ov-file#action-buttons) to build, load and debug the example on the hardware.

## Examples Description
| Example name                              | Description   |
|---                                        |---            |

> | [Hello](Hello.csolution.yml)              | The application module [`hello.c`](.\hello.c) prints "Hello World" with count value to the UART output. It is configured for [Arm Virtual Hardware - Fixed Virtual Platforms](https://arm-software.github.io/AVH/main/simulation/html/index.html) (AVH FVP), but it is easy to re-target it to hardware that provides a [CMSIS Driver:USART] (https://arm-software.github.io/CMSIS_6/latest/Driver/group__usart__interface__gr.html). |

Reviewer: ```suggestion
| [Hello](Hello.csolution.yml)              | The application module [`hello.c`](.\hello.c) prints "Hello World" with count value to the UART output. It is configured for [Arm Fixed Virtual Platforms](https://arm-software.github.io/AVH/main/simulation/html/index.html) (Arm FVP), but it is easy to re-target it to hardware that provides a [CMSIS Driver:USART] (https://arm-software.github.io/CMSIS_6/latest/Driver/group__usart__interface__gr.html). |
```

The workflow allows to build and test the application on different host systems, for example local development computers that are Windows based and CI systems that run Linux.

## Prerequisites for commandline build and execution

> - The required tools are installed from [ARM Tools Artifactory](https://www.keil.arm.com/artifacts/) using [vcpkg](https://learn.arm.com/learning-paths/microcontrollers/vcpkg-tool-installation/installation/).

Reviewer: ```suggestion
- The required tools are installed from [Arm Tools Artifactory](https://www.keil.arm.com/artifacts/) using [vcpkg](https://learn.arm.com/learning-paths/microcontrollers/vcpkg-tool-installation/installation/).
```

> - Depending on the local development computers, the `--simlimit 60` exceeds to execute the full test run. If this is the case increase the value, i.e. to `--simlimit 200`.
> - Some FVP models simulate very fast and may not synchronize with the Terminal program. In this case use `--quantum 100` to slow down to view the Terminal output.

## Continuous Integration (CI)

The underlying build system of [Keil Studio](https://www.keil.arm.com/) uses the [CMSIS-Toolbox](https://open-cmsis-pack.github.io/cmsis-toolbox/) and CMake. [CI](https://en.wikipedia.org/wiki/Continuous_integration) is effectively supported with:

- Tool installation based on a single [`vcpkg-configuration.json`](./vcpkg-configuration.json) file for desktop and CI environments.
- CMSIS solution files (`*.csolution.yml`) that enable seamless builds in CI, for example using GitHub actions.

> | <div style="width:150px"> CI Workflow </div>    | Description |

Reviewer: Why is there a <div> in this header?

|---                                              |--- |

> | [hello-ci.yml](/.github/workflows/hello-ci.yml) | It uses the same commands for build and execute, except that the tools use parameters from the [test matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs). The test matrix validates the application with GCC and Arm Compiler using a `Debug` and `Release` build. It builds and runs the application across the different [Arm Cortex-M processors](https://www.arm.com/products/silicon-ip-cpu?families=cortex-m) and various [Arm Corstone sub-systems](https://www.arm.com/products/silicon-ip-subsystems). It total it validates that 56 different variants execute correct on AVH FVP simulation models that represent a typical implementation of an Arm processor. For the execution on the AVH FVP models, the UART ouput is redirected to a log file using the parameter **`-C`**. This output is checked for correctness. |

Reviewer: ```suggestion
| [hello-ci.yml](/.github/workflows/hello-ci.yml) | It uses the same commands for build and execute, except that the tools use parameters from the [test matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs). The test matrix validates the application with GCC and Arm Compiler for Embedded using a `Debug` and `Release` build. It builds and runs the application across the different [Arm Cortex-M processors](https://www.arm.com/products/silicon-ip-cpu?families=cortex-m) and various [Arm Corstone sub-systems](https://www.arm.com/products/silicon-ip-subsystems). In total it validates that 56 different variants execute correctly on Arm FVP simulation models that represent a typical implementation of an Arm processor. For the execution on the Arm FVP models, the UART ouput is redirected to a log file using the parameter **`-C`**. This output is checked for correctness. |
```
